Checks for security posture on privileged assets identified by EntraOps and XSPM

Cloud-Architekt opened the pull request:
### Description

Add tests for privileged identities (identified by EntraOps and Critical Asset Management) to check exposure due weak security configurations.

- Implemented test MT.1077 to check that app registrations with privileged API permissions have no owners.
- Implemented test MT.1078 to ensure app registrations with highly privileged directory roles do not have owners.
- Implemented test MT.1079 to verify that privileged API permissions on service principals are not unused.
- Implemented test MT.1080 to check that credentials, tokens, or cookies from highly privileged users are not exposed on vulnerable endpoints.
- Implemented test MT.1081 to ensure hybrid users are not assigned Entra ID role assignments.

Additionally, created detailed documentation for each test, outlining the description, how to fix issues, and the risks associated with improper ownership and permissions.

### Your checklist for this pull request

🚨Please review the [guidelines for contributing](https://maester.dev/docs/contributing) to this repository.

- [x] Read the guidelines for contributions

#### PRs related to all code

- [x] Before you submit the PR, run the tests locally by running `/powershell/tests/pester.ps1`
- [x] After submitting, verify the tests are still passing on your PR in GitHub (the tests are run across all platforms).

#### PRs related to Maester Tests

If you are contributing a test or making updates to a test please verify these.

- [x] Check that the unique ID you want to use is not already picked. See [Pick next Maester test sequence number](https://github.com/maester365/maester/issues/697)
- [x] Add Test with your .ps1 and .md-file (make sure .ps1 file is encoded with "UTF8 with BOM") `.\powershell\public\maester...`
- [x] Add your new test-function to a corresponding .test.ps1 file `\tests\Maester..*.test.ps1`
- [x] Add your new test-function to 'FunctionsToExport'-List `.\powershell\Maester.psd1`
- [x] Add test Id, Severity and Title to "maester-config.json" file `.\tests\maester-config.json`
- [x] Add website documentation for test with name of "testId.md" `.\website\docs\tests\maester`
- [x] If Invoke-Maester parameters are changed, update the GitHub action to use the new parameters.

### Review

We will try to review your pull request as soon as possible. If you have any queries or need any help please jump on [Discord](https://discord.maester.dev/). We really appreciate your contributions!

<!-- While your wait for a review, why not try to spread some Maester love on social media? -->

💖 Thank you!

merill reviewed (changes requested):
See comments for review

SamErde commented:
Another interesting thought on the icons: if those icon reference functions will be used by multiple tests, they're great.

If they will only be called by the one test that references them, they could be referenced as a hash table within the test like:

```powershell
$ArtifactTypes = [hashtable]@{
    'PrimaryRefreshToken' = "🪙"
    'UserCookie' = "🍪"
    'UserAzureCliSecretData' = "🔑"
    'Default' = "ℹ️"
}

$ObjectType = 'UserCookie'
$SampleResult = "The icon for an $ObjectType is a $($ArtifactTypes[$ObjectType])"
```

Cloud-Architekt commented:
@merill : First of all many thanks for the feedback and suggestions to improve the check logic.
I've moved the logic to separate cmdlets as suggested. I've also added the "How to Fix" section and a short description to the markdown file for each cmdlet. Would this be in alignment with the usual convention?

The prerequisites for MDA App Governance and MDI are particularly needed for these complex checks because of the combined data collection between XSPM, raw posture data, and EntraOps. Therefore, I've added a description and a link to the Microsoft Docs for the affected checks.  

Unfortunately, all of the checks in the PR require those sources. However, in the future most XSPM tests will work without this dependency. Therefore, I would like to suggest keeping the details of the prerequisites in the check documentation and also adding them as a deep link in the skipped result detail message. What do you think?

Additionally, there is a dependency on MDI for retrieving eligible assignments from Microsoft Entra for XDR's `IdentityInfo`, as documented [here](https://learn.microsoft.com/en-us/defender-xdr/advanced-hunting-identityinfo-table). This information is required for all checks (except the unused permissions) to cover all types of privileges of the user or application owner for the Tiered Administration Level classification. Please let me know your thoughts on this and if there is any other feedback or ideas.

@SamErde : Thanks for the suggestion to improve readability by using switch. This looks much besser :)

merill commented:
Thanks so much for all the updates @Cloud-Architekt !!

Very excited for more tests like this to be built by the community.
